src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp

```diff
@@ -0,0 +1,257 @@
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SeismicityRateBase.cpp
+ */
+
+#include "SeismicityRateBase.hpp"
+
+#include "dataRepository/InputFlags.hpp"
+#include "mainInterface/GeosxState.hpp"
+#include "mesh/mpiCommunications/CommunicationTools.hpp"
+#include "physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.hpp"
+
+namespace geos
+{
+
+using namespace dataRepository;
+using namespace fields;
+using namespace constitutive;
+
+//START_SPHINX_INCLUDE_CONSTRUCTOR
+SeismicityRateBase::SeismicityRateBase( const string & name,
+                              Group * const parent ):
+  SolverBase( name, parent ),
+  m_stressSolver( nullptr )
+  {
+    this->registerWrapper( viewKeyStruct::stressSolverNameString(), &m_stressSolverName ).
+          setInputFlag( InputFlags::REQUIRED ).
+          setDescription( "Name of solver for computing stress" );
+    this->registerWrapper( viewKeyStruct::faultNormalString(), &m_faultNormal ).
+          setInputFlag( InputFlags::OPTIONAL ).
+          setDescription( "Fault normal direction" );
+    this->registerWrapper( viewKeyStruct::faultShearString(), &m_faultShear ).
+          setInputFlag( InputFlags::OPTIONAL ).
+          setDescription( "Fault shear direction" );
+    this->registerWrapper( viewKeyStruct::initialFaultNormalStressString(), &m_initialFaultNormalStress ).
+          setInputFlag( InputFlags::OPTIONAL ).
+          setDescription( "Initial normal stress" );
+    this->registerWrapper( viewKeyStruct::initialFaultShearStressString(), &m_initialFaultShearStress ).
+          setInputFlag( InputFlags::OPTIONAL ).
+          setDescription( "Initial shear stress" );
+  }
+//END_SPHINX_INCLUDE_CONSTRUCTOR
+
+//START_SPHINX_INCLUDE_REGISTERDATAONMESH
+void SeismicityRateBase::registerDataOnMesh( Group & meshBodies )
+{
+  SolverBase::registerDataOnMesh( meshBodies );
+
+  forDiscretizationOnMeshTargets( meshBodies, [&] ( string const &,
+                                                    MeshLevel & mesh,
+                                                    arrayView1d< string const > const & regionNames )
+  {
+    ElementRegionManager & elemManager = mesh.getElemManager();
+
+    elemManager.forElementSubRegions< ElementSubRegionBase >( regionNames,
+                                                              [&]( localIndex const,
+                                                                   ElementSubRegionBase & subRegion )
+    {
+      subRegion.registerField< inducedSeismicity::meanStress >( getName() ).
+        reference().resizeDimension< 1 >( 6 );
+
+      subRegion.registerField< inducedSeismicity::initialMeanNormalStress >( getName() );
+      subRegion.registerField< inducedSeismicity::initialMeanShearStress >( getName() );
+
+      subRegion.registerField< inducedSeismicity::meanNormalStress >( getName() );
```

Comment:
As a general comment, it seems to me like we are potentially storing lots of data arrays that could be computed on the fly.  For example, could we just add a function to the solid material model to return the mean normal and shear stresses as needed?  For the seismicity rate calculation itself, are we storing any temporary quantities that could be recomputed on the fly, or do we need all these arrays (R, h, logDenom, etc)?

Reply:
Also, perhaps we should discuss the terminology.  The normal stress here is the projected normal traction on the fault, averaged over quadrature points?  Perhaps we can call this `projectedNormalTraction` and `projectedShearTraction`?  At first I was thinking this was some measure of the volumetric and deviatory stress in the body.  If the nomenclature is clear to others, though, let's keep it.